src/Symfony/Component/Routing/Router.php

```diff
@@ -442,4 +444,21 @@ private function checkDeprecatedOption($key)
                 @trigger_error(sprintf('Option "%s" given to router %s is deprecated since Symfony 4.3.', $key, static::class), E_USER_DEPRECATED);
         }
     }
+
+    private static function getCompiledRoutes(string $path): array
+    {
+        if ([] === self::$cache && \function_exists('opcache_invalidate') && filter_var(ini_get('opcache.enable'), FILTER_VALIDATE_BOOLEAN) && (!\in_array(\PHP_SAPI, ['cli', 'phpdbg'], true) || filter_var(ini_get('opcache.enable_cli'), FILTER_VALIDATE_BOOLEAN))) {
+            self::$cache = null;
```

Comment:
What is the reason for disabling this memory cache when using OPCache ? Does OPCache fix the memory leak ?

Reply:
yes, opcache fixes the leak: the require data are plain arrays, which don't leak.